engine/runtime-compiler/src/test/java/org/enso/compiler/test/mock/VirtualFileSystem.java

```diff
@@ -0,0 +1,212 @@
+package org.enso.compiler.test.mock;
+
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileTime;
+import java.util.Arrays;
+import java.util.stream.Stream;
+import org.enso.filesystem.FileSystem;
+
+/**
+ * {@link FileSystem} implementation that keeps all the created files in memory. Does not access the
```

Comment:
The comment isn't really correct. This `FileSystem` implementation works on any `Path`...